test/libsolidity/SolidityNameAndTypeResolution.cpp

```diff
@@ -5067,15 +5066,14 @@ BOOST_AUTO_TEST_CASE(inline_assembly_storage_in_modifiers)
 			modifier m {
 				assembly {
 					x := 2
-					pop
 				}
 				_;
 			}
 			function f() m {
 			}
 		}
 	)";
-	CHECK_ERROR(text, DeclarationError, "");
+	CHECK_ERROR(text, DeclarationError, "Variable not found or variable not lvalue.");
```

Comment:
Please add tests about accessing constant state variables in both L and R value contexts.